ros_ethercat_model/include/ros_ethercat_model/ros_ethercat.hpp

```diff
@@ -439,7 +450,35 @@ class RosEthercat : public hardware_interface::RobotHW
     unlink(filename.c_str());
   }
 
+  void collect_diagnostics_loop()
+  {
+    collect_diagnostics_running_ = true;
+    ros::Rate diag_rate(1.0); // Send diagnostics at 1Hz
+    while (run_diagnostics_)
```

Comment:
I don't think it's necessary, as the destructor of RosEthercat will stop the diagnostics.
This is the same pattern used in the [RealtimePublisher](https://github.com/ros-controls/realtime_tools/blob/indigo-devel/include/realtime_tools/realtime_publisher.h#L184).